src/services/postgres/work_item.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Services
+  module Postgres
+    ##
+    # WorkItem Service for PostgreSQL
+    #
+    # Provides CRUD operations for the 'work_items' table using the Base service.
+    class WorkItem < Services::Postgres::Base
+      TABLE = :work_items
+
+      # Mapping of external IDs to internal IDs for relations.
+      RELATIONS = {
+        external_activity_id: {
+          table: :activities,
+          id_field: :activity_id,
+          external_field: :external_activity_id
+        },
+        external_project_id: {
+          table: :projects,
+          id_field: :project_id,
+          external_field: :external_project_id
+        }
+        # Here you can add more relations as needed
+      }.freeze
+
+      def insert(params)
+        params = params.dup
+        resolve_foreign_keys(params)
+        transaction { insert_item(TABLE, params) }
+      rescue StandardError => e
+        handle_error(e)
+      end
```

Comment:
I think we can make it cleaner if we reause the services you already added. We could import the `activity` and `project` services, and use their query method to search for the records. Maybe something like this:

```ruby
require_relative 'activity'

module Services
  module Postgres
    class WorkItem
       def insert
        ...
        activity_id = Activity.new(db).query(external_activity_id: params[:external_activity_id]).id # reuse the database connection of the work item instance!
        ...
       end
    end
  end
end
```
Let me know what you think and if you want to discuss it!